### PR TITLE
Change double quotes to single quotes and remove new line in error messages

### LINKED ISSUE #2353: Change double quotes to single quotes in error messages
Example
изображение

### LINKED ISSUE #2356: Allow parsing enum variable by native .NET type
> There's just no reason for any external values to be parsed as internal values.

My point is that for variables we cannot say for sure what is `internal` and `external` values.

> There's no way for json -- the transport of choice for graphql -- to carry an .NET enumeration type to be parsed at all. 

Forget about JSON. This is just one of the sources. Think about the fact that the variables can simply be created programmatically. In the end, this is the essence of the runtime - the runtime/platform on which the application is running.

> So if you allowed ParseValue to contain the internal value 1, you should also allow the literal to parse the value 1. 

Absolutely no. Literal is a part of GraphQL language. Variables are transport/platform specific. Also keep in mind that there may be no any transport at all.

> I don't mind allowing native types within ParseValue for Guid/Uri/DateTime/DateTimeOffset because you can consider the values as external values -- **in another format**. Just like receiving a double for a decimal graph type -- no problem, coerce the type. But parsing an enum against an internal list of values is not the same thing.

Why that argument about _just another format_ cannot be applied to enums ?

I suggest to add a new issue pointing to that thread and go on with this PR.

_Originally posted by @sungam3r in https://github.com/graphql-dotnet/graphql-dotnet/pull/2341#discussion_r588407993_

### PR DESCRIPTION
**Pull request by sungam3r**
fixes #2353

### CODE REVIEW

File: src/GraphQL.Tests/Bugs/Bug1205VeryLongInt.cs [hunk 1]
```diff
@@ -36,7 +36,7 @@ public void Very_Long_Number_In_Input_Should_Return_Error_Without_Data_For_Int()
             {
                 Errors = new ExecutionErrors
                 {
-                    new ValidationError(null, ArgumentsOfCorrectTypeError.NUMBER, "Argument \"in\" has invalid value 636474637870330463.\nExpected type \"Int\", found 636474637870330463.")
+                    new ValidationError(null, ArgumentsOfCorrectTypeError.NUMBER, "Argument 'in' has invalid value. Expected type 'Int', found 636474637870330463.")
```
Comment (sungam3r) [inline src/GraphQL.Tests/Bugs/Bug1205VeryLongInt.cs:39]: In new error messages value does not repeat twice.

File: src/GraphQL.Tests/Validation/ArgumentsOfCorrectTypeTests.cs [hunk 2]
```diff
@@ -543,11 +542,7 @@ public void list_value_incorrect_item_type()
             ShouldFailRule(_ =>
             {
                 _.Query = query;
-                Rule.badValue(_, "stringListArg", "[String]", "[\"one\", 2]", 3, 36,
-                    new[]
-                    {
-                        "In element #1: Expected type \"String\", found 2."
-                    });
+                Rule.badValue(_, "stringListArg", "[String]", "[\"one\", 2]", 3, 36, "In element #2: [Expected type 'String', found 2.]");
```
Comment (sungam3r) [inline src/GraphQL.Tests/Validation/ArgumentsOfCorrectTypeTests.cs:545]: Notice change in index. Was 1, became 2. This is due to bug in `IsValidLiteralValue` before:
```csharp
 return listValue.Values
                        .SelectMany(value =>
                            IsValidLiteralValue(ofType, value, schema)
                                .Select((err, index) => $"In element #{index + 1}: {err}")
                        )
                        .ToArray();
```

Index should be taken over "external" list, not over internal `string[]` array of errors.

File: src/GraphQL/GraphQLExtensions.cs [hunk 3]
```diff
@@ -283,12 +281,18 @@ public static string[] IsValidLiteralValue(this IGraphType type, IValue valueAst
 
                 if (valueAst is ListValue listValue)
                 {
-                    return listValue.Values
```
Comment (sungam3r) [inline src/GraphQL/GraphQLExtensions.cs:286]: No more bug here. No more LINQ. No more closures and allocations.

File: src/GraphQL/Validation/Errors/ArgumentsOfCorrectTypeError.cs [hunk 4]
```diff
@@ -14,16 +12,16 @@ public class ArgumentsOfCorrectTypeError : ValidationError
...
 
-        internal static string BadValueMessage(string argName, string value, IEnumerable<string> verboseErrors)
+        internal static string BadValueMessage(string argName, string verboseErrors)
         {
-            var message = verboseErrors != null ? $"\n{string.Join("\n", verboseErrors)}" : "";
-
-            return $"Argument \"{argName}\" has invalid value {value}.{message}";
+            return string.IsNullOrEmpty(verboseErrors)
+                ? $"Argument '{argName}' has invalid value."
+                : $"Argument '{argName}' has invalid value. {verboseErrors}";
```
Comment (Shane32) [inline src/GraphQL/Validation/Errors/ArgumentsOfCorrectTypeError.cs:24]: Simplify?  Either way is fine.
```suggestion
            return $"Argument '{argName}' has invalid value.{(string.IsNullOrEmpty(verboseErrors) ? null : " " + verboseErrors)}";
```

File: src/GraphQL/Validation/Errors/DefaultValuesOfCorrectTypeError.cs [hunk 5]
```diff
@@ -15,15 +14,16 @@ public class DefaultValuesOfCorrectTypeError : ValidationError
...
         }
 
-        internal static string BadValueForDefaultArgMessage(string varName, string type, string value, IEnumerable<string> verboseErrors)
+        internal static string BadValueForDefaultArgMessage(string varName, string type, string value, string verboseErrors)
         {
-            var message = verboseErrors != null ? "\n" + string.Join("\n", verboseErrors) : "";
-            return $"Variable \"{varName}\" of type \"{type}\" has invalid default value {value}.{message}";
+            return string.IsNullOrEmpty(verboseErrors)
+                ? $"Variable '{varName}' of type '{type}' has invalid default value '{value}'."
+                : $"Variable '{varName}' of type '{type}' has invalid default value '{value}'. {verboseErrors}";
```
Comment (Shane32) [inline src/GraphQL/Validation/Errors/DefaultValuesOfCorrectTypeError.cs:26]: Simplify?

File: src/GraphQL/Validation/Errors/NoUndefinedVariablesError.cs [hunk 6]
```diff
@@ -19,7 +19,7 @@ public NoUndefinedVariablesError(ValidationContext context, Operation node, Vari
 
         internal static string UndefinedVarMessage(string varName, string opName)
             => !string.IsNullOrWhiteSpace(opName)
-                ? $"Variable \"${varName}\" is not defined by operation \"{opName}\"."
-                : $"Variable \"${varName}\" is not defined.";
+                ? $"Variable '${varName}' is not defined by operation '{opName}'."
+                : $"Variable '${varName}' is not defined.";
```
Comment (Shane32) [inline src/GraphQL/Validation/Errors/NoUndefinedVariablesError.cs:23]: Simplify?

Comment (Shane32) [reply src/GraphQL/Validation/Errors/NoUndefinedVariablesError.cs:23]: Maybe not here...

File: src/GraphQL/Validation/Errors/ArgumentsOfCorrectTypeError.cs (hunk 4 above)
Comment (sungam3r) [reply src/GraphQL/Validation/Errors/ArgumentsOfCorrectTypeError.cs:24]: This option is more difficult to read and there are more allocations in it, although they are not on the hot path.

**Comment by Shane32** [comment]
Looks great!  What about errors ending in periods?  I know we have some of those.

File: src/GraphQL/Validation/Errors/NoUndefinedVariablesError.cs (hunk 6 above)
Comment (sungam3r) [reply src/GraphQL/Validation/Errors/NoUndefinedVariablesError.cs]: It is better to save readability, I inverted the condition.

**Comment by Shane32** [comment]
Newlines still present at SchemaTypes.cs:487 and SchemaTypes.cs:712 - want to catch those also?

```cs
                throw new InvalidOperationException($@"A loop has been detected while registering schema types.
There was an attempt to re-register '{namedType.FullName}' with instance of '{resolvedType.GetType().FullName}'.
Make sure that your ServiceProvider is configured correctly.");
```
```cs
                    throw new InvalidOperationException($@"Unable to register GraphType '{type.GetType().FullName}' with the name '{typeName}'.
The name '{typeName}' is already registered to '{existingGraphType.GetType().FullName}'. Check your schema configuration.");
```

**Comment by sungam3r** [comment]
Is it bad?

**Comment by Shane32** [comment]
I just thought we were going to consolidate so all errors ended in a period?  Can be done later of course.

**Comment by sungam3r** [comment]
> Newlines still present at SchemaTypes.cs:487 and SchemaTypes.cs:712 - want to catch those also?

No, those are not execution errors serialized into response.

**Comment by Shane32** [comment]
Ok

**Comment by sungam3r** [comment]
> I just thought we were going to consolidate so all errors ended in a period?

🤔 How do I find them all?

**Review by Shane32 — APPROVED** [review]

**Comment by Shane32** [comment]
> > I just thought we were going to consolidate so all errors ended in a period?
> 
> 🤔 How do I find them all?

Yeah that's the hard part...

**Pull request merged**